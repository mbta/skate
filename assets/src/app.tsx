// We need to import the CSS so that webpack will load it.
// The MiniCssExtractPlugin is used to separate it out into
// its own CSS file.
declare function require(name: string): string
// tslint:disable-next-line
require("../css/app.scss")

// webpack automatically bundles all modules in your
// entry points. Those entry points can be configured
// in "webpack.config.js".
//
// Import dependencies
//
import "@babel/polyfill"
import "leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.webpack.css" // see https://github.com/Leaflet/Leaflet/issues/4968#issuecomment-483402699
import "leaflet/dist/leaflet.css"
import "phoenix_html"
import * as React from "react"
import ReactDOM from "react-dom"
import AppStateWrapper from "./components/appStateWrapper"

if (window.FS && window.username) {
  window.FS.identify(window.username, { displayName: window.username })
}

// Import local files
//
// Local files can be imported directly using relative paths, for example:
// import socket from "./socket"

ReactDOM.render(<AppStateWrapper />, document.getElementById("app"))
