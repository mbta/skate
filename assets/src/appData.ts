const appData = (): DOMStringMap | undefined => {
  const dataEl = document.getElementById("app")

  if (!dataEl) {
    return undefined
  }

  return dataEl.dataset
}

export default appData
