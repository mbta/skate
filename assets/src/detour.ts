import { LatLngLiteral } from "leaflet"
import { Shape, ShapePoint } from "./schedule"

export interface DetourShape {
  coordinates: ShapePoint[]
  directions: {
    instruction: string
  }[]
}

export interface OriginalRoute {
  routeName: string
  routeDescription: string
  routeOrigin: string
  routeDirection: string
  shape: Shape
  center: LatLngLiteral
  zoom: number
}
