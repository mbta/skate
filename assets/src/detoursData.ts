import { array, Infer, number, string, type } from "superstruct"
import { DraftDetours } from "./models/detours"

export const DraftDetoursData = array(
  type({
    route: string(),
    direction: string(),
    name: string(),
    intersection: string(),
    active_since: number(),
  })
)
export type DraftDetoursData = Infer<typeof DraftDetoursData>

export const draftDetoursFromData = (
  draftDetoursData: DraftDetoursData
): DraftDetours =>
  draftDetoursData.map((draft) => ({
    route: draft.route,
    direction: draft.direction,
    name: draft.name,
    intersection: draft.intersection,
    activeSince: draft.active_since,
  }))
