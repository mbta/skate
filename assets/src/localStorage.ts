export const saveState = (stateKey: string, state: object) => {
  const serializedState = JSON.stringify(state)
  localStorage.setItem(stateKey, serializedState)
}

export const loadState = (stateKey: string): object | undefined => {
  const serializedState = localStorage.getItem(stateKey)

  if (serializedState === null) {
    return undefined
  }

  return JSON.parse(serializedState)
}
