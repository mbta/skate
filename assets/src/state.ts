import { Dispatch as ReactDispatch } from "react"
import {
  emptyLadderCrowdingTogglesByRouteId,
  LadderCrowdingToggles,
  toggleLadderCrowdingForRoute,
} from "./models/ladderCrowdingToggle"
import {
  emptyLadderDirectionsByRouteId,
  flipLadderDirectionForRoute,
  LadderDirections,
} from "./models/ladderDirection"
import { Notification, RunId, VehicleId } from "./realtime.d"
import { RouteId } from "./schedule.d"
import {
  Action as SearchAction,
  initialSearchPageState,
  reducer as searchReducer,
  SearchPageState,
} from "./state/searchPageState"
import {
  defaultUserSettings,
  UserSettings,
  VehicleLabelSetting,
} from "./userSettings"

export interface State {
  pickerContainerIsVisible: boolean
  searchPageState: SearchPageState
  selectedRouteIds: RouteId[]
  ladderDirections: LadderDirections
  ladderCrowdingToggles: LadderCrowdingToggles
  selectedShuttleRouteIds: RouteId[]
  selectedShuttleRunIds: RunId[] | "all"
  selectedVehicleId?: VehicleId
  userSettings: UserSettings
  selectedNotification?: Notification
}

export const initialState: State = {
  pickerContainerIsVisible: true,
  searchPageState: initialSearchPageState,
  selectedRouteIds: [],
  ladderDirections: emptyLadderDirectionsByRouteId,
  ladderCrowdingToggles: emptyLadderCrowdingTogglesByRouteId,
  selectedShuttleRouteIds: [],
  selectedShuttleRunIds: "all",
  selectedVehicleId: undefined,
  userSettings: defaultUserSettings,
  selectedNotification: undefined,
}

interface SelectRouteAction {
  type: "SELECT_ROUTE"
  payload: {
    routeId: RouteId
  }
}

export const selectRoute = (routeId: RouteId): SelectRouteAction => ({
  type: "SELECT_ROUTE",
  payload: { routeId },
})

export interface DeselectRouteAction {
  type: "DESELECT_ROUTE"
  payload: {
    routeId: RouteId
  }
}

export const deselectRoute = (routeId: RouteId): DeselectRouteAction => ({
  type: "DESELECT_ROUTE",
  payload: { routeId },
})

interface FlipLadderAction {
  type: "FLIP_LADDER"
  payload: {
    routeId: RouteId
  }
}

export const flipLadder = (routeId: RouteId): FlipLadderAction => ({
  type: "FLIP_LADDER",
  payload: { routeId },
})

interface ToggleLadderCrowdingAction {
  type: "TOGGLE_LADDER_CROWDING"
  payload: { routeId: RouteId }
}

export const toggleLadderCrowding = (
  routeId: RouteId
): ToggleLadderCrowdingAction => ({
  type: "TOGGLE_LADDER_CROWDING",
  payload: { routeId },
})

interface SelectShuttleRunAction {
  type: "SELECT_SHUTTLE_RUN"
  payload: {
    runId: RunId
  }
}

export const selectShuttleRun = (runId: RunId): SelectShuttleRunAction => ({
  type: "SELECT_SHUTTLE_RUN",
  payload: {
    runId,
  },
})

interface DeselectShuttleRunAction {
  type: "DESELECT_SHUTTLE_RUN"
  payload: {
    runId: RunId
  }
}

export const deselectShuttleRun = (runId: RunId): DeselectShuttleRunAction => ({
  type: "DESELECT_SHUTTLE_RUN",
  payload: {
    runId,
  },
})

interface SelectAllShuttleRunsAction {
  type: "SELECT_ALL_SHUTTLE_RUNS"
}

export const selectAllShuttleRuns = (): SelectAllShuttleRunsAction => ({
  type: "SELECT_ALL_SHUTTLE_RUNS",
})

interface DeselectAllShuttleRunsAction {
  type: "DESELECT_ALL_SHUTTLE_RUNS"
}

export const deselectAllShuttleRuns = (): DeselectAllShuttleRunsAction => ({
  type: "DESELECT_ALL_SHUTTLE_RUNS",
})

interface SelectShuttleRouteAction {
  type: "SELECT_SHUTTLE_ROUTE"
  payload: {
    routeId: RouteId
  }
}

export const selectShuttleRoute = (
  routeId: RouteId
): SelectShuttleRouteAction => ({
  type: "SELECT_SHUTTLE_ROUTE",
  payload: {
    routeId,
  },
})

interface DeselectShuttleRouteAction {
  type: "DESELECT_SHUTTLE_ROUTE"
  payload: {
    routeId: RouteId
  }
}

export const deselectShuttleRoute = (
  routeId: RouteId
): DeselectShuttleRouteAction => ({
  type: "DESELECT_SHUTTLE_ROUTE",
  payload: {
    routeId,
  },
})

export interface SelectVehicleAction {
  type: "SELECT_VEHICLE"
  payload: {
    vehicleId: VehicleId
  }
}

export const selectVehicle = (vehicleId: VehicleId): SelectVehicleAction => ({
  type: "SELECT_VEHICLE",
  payload: { vehicleId },
})

export interface DeselectVehicleAction {
  type: "DESELECT_VEHICLE"
}

export const deselectVehicle = (): DeselectVehicleAction => ({
  type: "DESELECT_VEHICLE",
})

interface TogglePickerContainerAction {
  type: "TOGGLE_PICKER_CONTAINER"
}

export const togglePickerContainer = (): TogglePickerContainerAction => ({
  type: "TOGGLE_PICKER_CONTAINER",
})

interface SetLadderVehicleLabelSettingAction {
  type: "SET_LADDER_VEHICLE_LABEL_SETTING"
  payload: {
    ladderVehicleLabel: VehicleLabelSetting
  }
}

export const setLadderVehicleLabelSetting = (
  ladderVehicleLabel: VehicleLabelSetting
): SetLadderVehicleLabelSettingAction => ({
  type: "SET_LADDER_VEHICLE_LABEL_SETTING",
  payload: {
    ladderVehicleLabel,
  },
})

interface SetShuttleVehicleLabelSettingAction {
  type: "SET_MAP_VEHICLE_LABEL_SETTING"
  payload: {
    shuttleVehicleLabel: VehicleLabelSetting
  }
}

export const setShuttleVehicleLabelSetting = (
  shuttleVehicleLabel: VehicleLabelSetting
): SetShuttleVehicleLabelSettingAction => ({
  type: "SET_MAP_VEHICLE_LABEL_SETTING",
  payload: {
    shuttleVehicleLabel,
  },
})

interface SetNotificationAction {
  type: "SET_NOTIFICATION"
  payload: {
    selectedNotification?: Notification
  }
}

export const setNotification = (
  selectedNotification?: Notification
): SetNotificationAction => ({
  type: "SET_NOTIFICATION",
  payload: {
    selectedNotification,
  },
})

type Action =
  | SelectRouteAction
  | DeselectRouteAction
  | FlipLadderAction
  | ToggleLadderCrowdingAction
  | SelectShuttleRunAction
  | DeselectShuttleRunAction
  | SelectAllShuttleRunsAction
  | DeselectAllShuttleRunsAction
  | SelectShuttleRouteAction
  | DeselectShuttleRouteAction
  | SelectVehicleAction
  | DeselectVehicleAction
  | TogglePickerContainerAction
  | SetLadderVehicleLabelSettingAction
  | SetShuttleVehicleLabelSettingAction
  | SearchAction
  | SetNotificationAction

export type Dispatch = ReactDispatch<Action>

export type Reducer = (state: State, action: Action) => State

const shuttleRunIdsList = (selectedShuttleRunIds: RunId[] | "all"): RunId[] =>
  selectedShuttleRunIds === "all" ? [] : selectedShuttleRunIds

const pickerContainerIsVisibleReducer = (
  state: boolean,
  action: Action
): boolean => {
  switch (action.type) {
    case "TOGGLE_PICKER_CONTAINER":
      return !state
    default:
      return state
  }
}

const selectedRouteIdsReducer = (
  state: RouteId[],
  action: Action
): RouteId[] => {
  switch (action.type) {
    case "SELECT_ROUTE":
      return [...state, action.payload.routeId]
    case "DESELECT_ROUTE":
      return state.filter((id) => id !== action.payload.routeId)
    default:
      return state
  }
}

const ladderDirectionsReducer = (
  state: LadderDirections,
  action: Action
): LadderDirections => {
  switch (action.type) {
    case "FLIP_LADDER":
      const routeId = action.payload.routeId
      return flipLadderDirectionForRoute(state, routeId)
    default:
      return state
  }
}

const ladderCrowdingTogglesReducer = (
  state: LadderCrowdingToggles,
  action: Action
): LadderCrowdingToggles => {
  switch (action.type) {
    case "TOGGLE_LADDER_CROWDING":
      const routeId = action.payload.routeId
      return toggleLadderCrowdingForRoute(state, routeId)
    default:
      return state
  }
}

const selectedShuttleRouteIdsReducer = (
  state: RouteId[],
  action: Action
): RouteId[] => {
  switch (action.type) {
    case "SELECT_SHUTTLE_ROUTE":
      return [...state, action.payload.routeId]
    case "DESELECT_SHUTTLE_ROUTE":
      return state.filter((id) => id !== action.payload.routeId)
    default:
      return state
  }
}

const selectedShuttleRunIdsReducer = (
  state: RunId[] | "all",
  action: Action
): RunId[] | "all" => {
  switch (action.type) {
    case "SELECT_SHUTTLE_RUN":
      return [...shuttleRunIdsList(state), action.payload.runId]
    case "DESELECT_SHUTTLE_RUN":
      return shuttleRunIdsList(state).filter(
        (id) => id !== action.payload.runId
      )
    case "SELECT_ALL_SHUTTLE_RUNS":
      return "all"
    case "DESELECT_ALL_SHUTTLE_RUNS":
      return []
    default:
      return state
  }
}

const selectedVehicleIdReducer = (
  state: VehicleId | undefined,
  action: Action
): VehicleId | undefined => {
  switch (action.type) {
    case "SELECT_VEHICLE":
      return action.payload.vehicleId
    case "DESELECT_VEHICLE":
    case "SET_NOTIFICATION":
      return undefined
    default:
      return state
  }
}

const userSettingsReducer = (
  state: UserSettings,
  action: Action
): UserSettings => {
  switch (action.type) {
    case "SET_LADDER_VEHICLE_LABEL_SETTING":
      return {
        ...state,
        ladderVehicleLabel: action.payload.ladderVehicleLabel,
      }
    case "SET_MAP_VEHICLE_LABEL_SETTING":
      return {
        ...state,
        shuttleVehicleLabel: action.payload.shuttleVehicleLabel,
      }
    default:
      return state
  }
}

const selectedNotificationReducer = (
  state: Notification | undefined,
  action: Action
): Notification | undefined => {
  switch (action.type) {
    case "SELECT_VEHICLE":
    case "DESELECT_VEHICLE":
      return undefined
    case "SET_NOTIFICATION":
      return action.payload.selectedNotification
    default:
      return state
  }
}

export const reducer = (state: State, action: Action): State => ({
  pickerContainerIsVisible: pickerContainerIsVisibleReducer(
    state.pickerContainerIsVisible,
    action
  ),
  searchPageState: searchReducer(state.searchPageState, action as SearchAction),
  selectedRouteIds: selectedRouteIdsReducer(state.selectedRouteIds, action),
  ladderDirections: ladderDirectionsReducer(state.ladderDirections, action),
  ladderCrowdingToggles: ladderCrowdingTogglesReducer(
    state.ladderCrowdingToggles,
    action
  ),
  selectedShuttleRouteIds: selectedShuttleRouteIdsReducer(
    state.selectedShuttleRouteIds,
    action
  ),
  selectedShuttleRunIds: selectedShuttleRunIdsReducer(
    state.selectedShuttleRunIds,
    action
  ),
  selectedVehicleId: selectedVehicleIdReducer(state.selectedVehicleId, action),
  userSettings: userSettingsReducer(state.userSettings, action),
  selectedNotification: selectedNotificationReducer(
    state.selectedNotification,
    action
  ),
})
