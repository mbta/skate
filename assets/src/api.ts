import "whatwg-fetch"
import appData from "./appData"
import { Block, Run } from "./minischedule"
import { reload } from "./models/browser"
import { blockFromData, runFromData } from "./models/minischeduleData"
import { swingsFromData } from "./models/swingsData"
import { NotificationId, NotificationState } from "./realtime.d"
import { RouteSettings } from "./routeSettings"
import {
  DirectionName,
  Route,
  RouteId,
  Shape,
  Swing,
  Timepoint,
  TripId,
} from "./schedule.d"

export interface RouteData {
  id: string
  direction_names: {
    "0": DirectionName
    "1": DirectionName
  }
  name: string
}

const checkResponseStatus = (response: Response) => {
  if (response.status === 200) {
    return response
  }

  if (Math.floor(response.status / 100) === 3 || response.status === 403) {
    // If the API sends us a redirect or forbidden, the user needs to
    // re-authenticate. Reload to go through the auth flow again.
    reload(true)
  }

  throw new Error(`Response error: ${response.status}`)
}

const parseJson = (response: Response) => response.json()

export const apiCall = <T>({
  url,
  parser,
  defaultResult,
}: {
  url: string
  parser: (data: any) => T
  defaultResult?: T
}): Promise<T> =>
  fetch(url)
    .then(checkResponseStatus)
    .then(parseJson)
    .then(({ data: data }: { data: any }) => parser(data))
    .catch((error) => {
      if (defaultResult === undefined) {
        throw error
      } else {
        return defaultResult
      }
    })

export const parseRouteData = ({
  id,
  direction_names,
  name,
}: RouteData): Route => ({
  id,
  directionNames: direction_names,
  name,
})

const parseRoutesData = (routesData: RouteData[]): Route[] =>
  routesData.map(parseRouteData)

export const fetchRoutes = (): Promise<Route[]> =>
  apiCall({
    url: "/api/routes",
    parser: parseRoutesData,
  })

export const fetchShapeForRoute = (routeId: RouteId): Promise<Shape[]> =>
  apiCall({
    url: `/api/shapes/route/${routeId}`,
    parser: (shapes: Shape[]) => shapes,
    defaultResult: [],
  })

export const fetchShapeForTrip = (tripId: TripId): Promise<Shape | null> =>
  apiCall({
    url: `/api/shapes/trip/${tripId}`,
    parser: (shape: Shape | null) => shape,
    defaultResult: null,
  })

export const fetchShuttleRoutes = (): Promise<Route[]> =>
  apiCall({
    url: "/api/shuttles",
    parser: parseRoutesData,
    defaultResult: [],
  })

export const fetchTimepointsForRoute = (
  routeId: RouteId
): Promise<Timepoint[]> =>
  apiCall({
    url: `/api/routes/${routeId}`,
    parser: (timepoints: Timepoint[]) => timepoints,
    defaultResult: [],
  })

export const fetchMinischeduleRun = (tripId: TripId): Promise<Run | null> =>
  apiCall({
    url: `/api/minischedule/run/${tripId}`,
    parser: nullableParser(runFromData),
    defaultResult: null,
  })

export const fetchMinischeduleBlock = (tripId: TripId): Promise<Block | null> =>
  apiCall({
    url: `/api/minischedule/block/${tripId}`,
    parser: nullableParser(blockFromData),
    defaultResult: null,
  })

export const fetchNearestIntersection = (
  latitude: number,
  longitude: number
): Promise<string | null> =>
  apiCall({
    url: `/api/intersection?latitude=${latitude}&longitude=${longitude}`,
    parser: nullableParser((intersection: string) => intersection),
    defaultResult: null,
  })

export const fetchSwings = (): Promise<Swing[] | null> =>
  apiCall({
    url: "/api/swings",
    parser: nullableParser(swingsFromData),
    defaultResult: null,
  })

export const putNotificationReadState = (
  newReadState: NotificationState,
  notificationIds: NotificationId[]
): void => {
  const url = `/api/notification_read_state?new_state=${newReadState}&notification_ids=${notificationIds.join(
    ","
  )}`
  fetch(url, {
    method: "PUT",
    headers: {
      "Content-Type": "application/json",
      "x-csrf-token": getCsrfToken(),
    },
  })
}

export const putUserSetting = (field: string, value: string): void => {
  const url = `/api/user_settings?field=${field}&value=${value}`
  fetch(url, {
    method: "PUT",
    headers: {
      "Content-Type": "application/json",
      "x-csrf-token": getCsrfToken(),
    },
  })
}

export const putRouteSettings = (routeSettings: RouteSettings): void => {
  fetch("/api/route_settings", {
    method: "PUT",
    headers: {
      "Content-Type": "application/json",
      "x-csrf-token": getCsrfToken(),
    },
    body: JSON.stringify(routeSettings),
  })
}

const getCsrfToken = (): string => appData()?.csrfToken || ""

export const nullableParser =
  <Data, T>(parser: (data: Data) => T): ((data: Data | null) => T | null) =>
  (data: Data | null) =>
    data === null ? null : parser(data)
