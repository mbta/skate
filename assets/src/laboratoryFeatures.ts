import appData from "./appData"

const featureIsEnabled = (key: string): boolean => {
  const data = appData()

  if (data === undefined || data.laboratoryFeatures === undefined) {
    return false
  }

  const features = JSON.parse(data.laboratoryFeatures)
  if (features[key] === undefined) {
    return false
  }

  return features[key]
}

export default featureIsEnabled
