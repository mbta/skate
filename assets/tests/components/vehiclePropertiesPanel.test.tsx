import { mount } from "enzyme"
import React from "react"
import renderer from "react-test-renderer"
import VehiclePropertiesPanel, {
  formatRouteVariant,
} from "../../src/components/vehiclePropertiesPanel"
import DispatchProvider from "../../src/providers/dispatchProvider"
import { Vehicle } from "../../src/skate"
import { deselectVehicle } from "../../src/state"

const vehicle: Vehicle = {
  id: "v1",
  label: "v1-label",
  run_id: "run-1",
  timestamp: 123,
  latitude: 0,
  longitude: 0,
  direction_id: 0,
  route_id: "r1",
  trip_id: "t1",
  headsign: "headsign",
  via_variant: "4",
  operator_id: "op1",
  operator_name: "SMITH",
  stop_status: {
    status: "in_transit_to",
    stop_id: "s1",
    stop_name: "Stop Name",
  },
  timepoint_status: {
    timepoint_id: "tp1",
    fraction_until_timepoint: 0.5,
  },
  route_status: "on_route",
}

describe("VehiclePropertiesPanel", () => {
  test("renders a vehicle properties panel", () => {
    const tree = renderer
      .create(<VehiclePropertiesPanel selectedVehicle={vehicle} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test("clicking the X close button deselects the vehicle", () => {
    const mockDispatch = jest.fn()

    const wrapper = mount(
      <DispatchProvider dispatch={mockDispatch}>
        <VehiclePropertiesPanel selectedVehicle={vehicle} />
      </DispatchProvider>
    )
    wrapper
      .find(".m-vehicle-properties-panel__header .m-close-button")
      .simulate("click")

    expect(mockDispatch).toHaveBeenCalledWith(deselectVehicle())
  })

  test("clicking the 'Close vehicle properties' button deselects the vehicle", () => {
    const mockDispatch = jest.fn()

    const wrapper = mount(
      <DispatchProvider dispatch={mockDispatch}>
        <VehiclePropertiesPanel selectedVehicle={vehicle} />
      </DispatchProvider>
    )
    wrapper.find(".m-vehicle-properties-panel__close").simulate("click")

    expect(mockDispatch).toHaveBeenCalledWith(deselectVehicle())
  })
})

describe("formatRouteVariant", () => {
  test("has variant and headsign", () => {
    expect(formatRouteVariant("39", "X", "Forest Hills")).toEqual(
      "39_X Forest Hills"
    )
  })

  test("missing variant and headsign", () => {
    expect(formatRouteVariant("39", null, null)).toEqual("39_")
  })

  test("doesn't show underscore variant character", () => {
    expect(formatRouteVariant("39", "_", null)).toEqual("39_")
  })
})
