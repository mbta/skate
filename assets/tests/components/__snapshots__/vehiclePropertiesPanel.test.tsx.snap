// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehiclePropertiesPanel renders a vehicle properties panel 1`] = `
Array [
  <div
    className="m-vehicle-properties-panel"
    id="m-vehicle-properties-panel"
  >
    <div
      className="m-vehicle-properties-panel__header"
    >
      <div
        className="m-vehicle-properties-panel__label"
      >
        v1-label
      </div>
      <div
        className="m-vehicle-properties-panel__variant"
      >
        <div
          className="m-vehicle-properties-panel__inbound-outbound"
        >
          
        </div>
        <div
          className="m-vehicle-properties-panel__variant-name"
        >
          39_X Forest Hills
        </div>
        <div
          className="m-vehicle-properties-panel__schedule-adherence on-time"
        >
          <div
            className="m-vehicle-properties-panel__schedule-adherence-status-icon"
          >
            <svg
              height="10"
              width="10"
            >
              <circle
                cx="5"
                cy="5"
                r="5"
              />
            </svg>
          </div>
          <div
            className="m-vehicle-properties-panel__schedule-adherence-status-string"
          >
            on-time
          </div>
          <div
            className="m-vehicle-properties-panel__schedule-adherence-label"
          >
            (0 min early)
          </div>
        </div>
      </div>
      <button
        className="m-close-button"
        onClick={[Function]}
      >
        <span
          className=""
          dangerouslySetInnerHTML={
            Object {
              "__html": "SVG",
            }
          }
        />
      </button>
    </div>
    <table
      className="m-vehicle-properties-panel__vehicle-properties"
    >
      <tbody>
        <tr>
          <th
            className="m-vehicle-properties-panel__vehicle-property-label"
          >
            Run
          </th>
          <td
            className="m-vehicle-properties-panel__vehicle-property-value"
          >
            run-1
          </td>
        </tr>
        <tr>
          <th
            className="m-vehicle-properties-panel__vehicle-property-label"
          >
            Vehicle
          </th>
          <td
            className="m-vehicle-properties-panel__vehicle-property-value"
          >
            v1-label
          </td>
        </tr>
        <tr>
          <th
            className="m-vehicle-properties-panel__vehicle-property-label"
          >
            Operator
          </th>
          <td
            className="m-vehicle-properties-panel__vehicle-property-value"
          >
            SMITH
             #
            op1
          </td>
        </tr>
      </tbody>
    </table>
    <div
      className="m-vehicle-properties-panel__location"
    >
      <div
        className="m-vehicle-properties-panel__vehicle-property-label"
      >
        Next Stop
      </div>
      <div
        className="m-vehicle-properties-panel__vehicle-property-value"
      >
        Stop Name
      </div>
      <a
        className="m-vehicle-properties-panel__link"
        href="https://www.google.com/maps/dir/?api=1&destination=0,0&travelmode=driving"
      >
        Directions
      </a>
    </div>
    <button
      className="m-vehicle-properties-panel__close"
      onClick={[Function]}
    >
      Close
    </button>
  </div>,
  <div
    className="m-vehicle-properties-panel__modal-overlay"
    onClick={[Function]}
  />,
]
`;

exports[`VehiclePropertiesPanel renders for a late vehicle 1`] = `
Array [
  <div
    className="m-vehicle-properties-panel"
    id="m-vehicle-properties-panel"
  >
    <div
      className="m-vehicle-properties-panel__header"
    >
      <div
        className="m-vehicle-properties-panel__label"
      >
        v1-label
      </div>
      <div
        className="m-vehicle-properties-panel__variant"
      >
        <div
          className="m-vehicle-properties-panel__inbound-outbound"
        >
          
        </div>
        <div
          className="m-vehicle-properties-panel__variant-name"
        >
          39_X Forest Hills
        </div>
        <div
          className="m-vehicle-properties-panel__schedule-adherence late"
        >
          <div
            className="m-vehicle-properties-panel__schedule-adherence-status-icon"
          >
            <svg
              height="10"
              width="10"
            >
              <circle
                cx="5"
                cy="5"
                r="5"
              />
            </svg>
          </div>
          <div
            className="m-vehicle-properties-panel__schedule-adherence-status-string"
          >
            late
          </div>
          <div
            className="m-vehicle-properties-panel__schedule-adherence-label"
          >
            (6 min late)
          </div>
        </div>
      </div>
      <button
        className="m-close-button"
        onClick={[Function]}
      >
        <span
          className=""
          dangerouslySetInnerHTML={
            Object {
              "__html": "SVG",
            }
          }
        />
      </button>
    </div>
    <table
      className="m-vehicle-properties-panel__vehicle-properties"
    >
      <tbody>
        <tr>
          <th
            className="m-vehicle-properties-panel__vehicle-property-label"
          >
            Run
          </th>
          <td
            className="m-vehicle-properties-panel__vehicle-property-value"
          >
            run-1
          </td>
        </tr>
        <tr>
          <th
            className="m-vehicle-properties-panel__vehicle-property-label"
          >
            Vehicle
          </th>
          <td
            className="m-vehicle-properties-panel__vehicle-property-value"
          >
            v1-label
          </td>
        </tr>
        <tr>
          <th
            className="m-vehicle-properties-panel__vehicle-property-label"
          >
            Operator
          </th>
          <td
            className="m-vehicle-properties-panel__vehicle-property-value"
          >
            SMITH
             #
            op1
          </td>
        </tr>
      </tbody>
    </table>
    <div
      className="m-vehicle-properties-panel__location"
    >
      <div
        className="m-vehicle-properties-panel__vehicle-property-label"
      >
        Next Stop
      </div>
      <div
        className="m-vehicle-properties-panel__vehicle-property-value"
      >
        Stop Name
      </div>
      <a
        className="m-vehicle-properties-panel__link"
        href="https://www.google.com/maps/dir/?api=1&destination=0,0&travelmode=driving"
      >
        Directions
      </a>
    </div>
    <button
      className="m-vehicle-properties-panel__close"
      onClick={[Function]}
    >
      Close
    </button>
  </div>,
  <div
    className="m-vehicle-properties-panel__modal-overlay"
    onClick={[Function]}
  />,
]
`;

exports[`VehiclePropertiesPanel renders for an early vehicle 1`] = `
Array [
  <div
    className="m-vehicle-properties-panel"
    id="m-vehicle-properties-panel"
  >
    <div
      className="m-vehicle-properties-panel__header"
    >
      <div
        className="m-vehicle-properties-panel__label"
      >
        v1-label
      </div>
      <div
        className="m-vehicle-properties-panel__variant"
      >
        <div
          className="m-vehicle-properties-panel__inbound-outbound"
        >
          
        </div>
        <div
          className="m-vehicle-properties-panel__variant-name"
        >
          39_X Forest Hills
        </div>
        <div
          className="m-vehicle-properties-panel__schedule-adherence early"
        >
          <div
            className="m-vehicle-properties-panel__schedule-adherence-status-icon"
          >
            <svg
              height="10"
              width="10"
            >
              <circle
                cx="5"
                cy="5"
                r="5"
              />
            </svg>
          </div>
          <div
            className="m-vehicle-properties-panel__schedule-adherence-status-string"
          >
            early
          </div>
          <div
            className="m-vehicle-properties-panel__schedule-adherence-label"
          >
            (2 min early)
          </div>
        </div>
      </div>
      <button
        className="m-close-button"
        onClick={[Function]}
      >
        <span
          className=""
          dangerouslySetInnerHTML={
            Object {
              "__html": "SVG",
            }
          }
        />
      </button>
    </div>
    <table
      className="m-vehicle-properties-panel__vehicle-properties"
    >
      <tbody>
        <tr>
          <th
            className="m-vehicle-properties-panel__vehicle-property-label"
          >
            Run
          </th>
          <td
            className="m-vehicle-properties-panel__vehicle-property-value"
          >
            run-1
          </td>
        </tr>
        <tr>
          <th
            className="m-vehicle-properties-panel__vehicle-property-label"
          >
            Vehicle
          </th>
          <td
            className="m-vehicle-properties-panel__vehicle-property-value"
          >
            v1-label
          </td>
        </tr>
        <tr>
          <th
            className="m-vehicle-properties-panel__vehicle-property-label"
          >
            Operator
          </th>
          <td
            className="m-vehicle-properties-panel__vehicle-property-value"
          >
            SMITH
             #
            op1
          </td>
        </tr>
      </tbody>
    </table>
    <div
      className="m-vehicle-properties-panel__location"
    >
      <div
        className="m-vehicle-properties-panel__vehicle-property-label"
      >
        Next Stop
      </div>
      <div
        className="m-vehicle-properties-panel__vehicle-property-value"
      >
        Stop Name
      </div>
      <a
        className="m-vehicle-properties-panel__link"
        href="https://www.google.com/maps/dir/?api=1&destination=0,0&travelmode=driving"
      >
        Directions
      </a>
    </div>
    <button
      className="m-vehicle-properties-panel__close"
      onClick={[Function]}
    >
      Close
    </button>
  </div>,
  <div
    className="m-vehicle-properties-panel__modal-overlay"
    onClick={[Function]}
  />,
]
`;
